opendrift/models/basemodel.py

```diff
@@ -5188,94 +5201,74 @@ def add_halo_readers(self):
-                                'baseline',
-                                '-vf',
-                                'crop=trunc(iw/2)*2:trunc(ih/2)*2',  # cropping 1 pixel if not even
-                                '-pix_fmt',
-                                'yuv420p',
-                                '-an'
-                            ])
-                    elif fastwriter:
-                        FFwriter = PunkFFMpegWriter(
-                            fps=fps,
-                            codec='libx264',
-                            bitrate=1800,
-                            extra_args=[
-                                '-profile:v',
-                                'baseline',
-                                '-vf',
-                                'crop=trunc(iw/2)*2:trunc(ih/2)*2',  # cropping 1 pixel if not even
-                                '-pix_fmt',
-                                'yuv420p',
-                                '-an'
-                            ])
-                    anim.save(filename, writer=FFwriter)
-                except Exception as e:
-                    logger.info(e)
-                    anim.save(filename, fps=fps)
-                    logger.warning('Animation might not be HTML5 compatible.')
+    def __save_animation__(self, fig, plot_timestep, filename, frames, fps):
 
-        except Exception as e:
-            logger.info('Could not save animation:')
-            logger.info(e)
-            logger.debug(traceback.format_exc())
+        if filename is None or 'sphinx_gallery' in sys.modules:
+            filename = self._sphinx_gallery_filename(stack_offset=3)
+
+        logger.info('Saving animation to ' + str(filename) + '...')
+
+        start_time = datetime.now()
+
+        if str(filename)[-4:] == '.gif':
+            writer = animation.PillowWriter(fps=fps)
+            # writer=animation.ImageMagickWriter(fps=fps)
+        elif str(filename)[-4:] == '.mp4':
+            writer = animation.FFMpegWriter(
+                fps=fps,
+                codec='libx264',
+                bitrate=1800,
+                extra_args=[
+                    '-profile:v',
+                    'baseline',
+                    '-vf',
+                    'crop=trunc(iw/2)*2:trunc(ih/2)*2',  # cropping 1 pixel if not even
+                    '-pix_fmt',
+                    'yuv420p',
+                    '-an'
+                ])
+
+        with writer.saving(fig, filename, None):
+            for i in range(frames):
+                plot_timestep(i)
+                writer.grab_frame()
 
         logger.debug(f"MPLBACKEND = {matplotlib.get_backend()}")
         logger.debug(f"DISPLAY = {os.environ.get('DISPLAY', 'None')}")
-        logger.debug(f"fastwriter: {fastwriter}")
+        logger.debug('Time to save animation: %s' %
+                     (datetime.now() - start_time))
 
-        if 'sphinx_gallery' in sys.modules:
-            plt.close()
+        plt.close()
```

Comment:
Did you have to remove the 'sphinx_gallery' conditionals? Is always GUI plotting always done using FuncAnimation?